libvast/src/plugin.cpp

```diff
@@ -380,6 +382,55 @@ store_plugin::make_store(system::accountant_actor accountant,
                                                  std::move(path), name());
 }
 
+// -- loader plugin -------------------------------------------------------------
+
+auto stdin_loader_plugin::make_loader(options, const bool_operator*) const
+  -> caf::expected<input_loader> {
+  return []() -> generator<chunk_ptr> {
+    auto in_buf = detail::fdinbuf(STDIN_FILENO, max_chunk_size);
+    in_buf.read_timeout() = read_timeout;
+    auto chunk_str = std::string{};
```

Comment:
I think this should not be a `std::string` to avoid assumptions about encoding, and to avoid having to also always ensure that a null-terminator exists (which is mandated by the standard). Generally, use vectors over strings for growable buffers, and use `std::byte` or `unsigned char` to avoid making assumptions about the encoding.

Reply:
Yeah, I forgot to switch out the `std::string` because it was easier to debug. I switched to `std::vector<std::byte>` in https://github.com/tenzir/vast/pull/2935/commits/ea29892b310fee9a14ce51876008f8093b946235